judge/views/problem.py

```diff
@@ -565,7 +565,12 @@ def problem_submit(request, problem=None, submission=None):
             else:
                 model = form.save()
 
+            # Create the SubmissionSource object
+            source = SubmissionSource(submission=model, source=form.cleaned_data['source'])
+            source.save()
```

Comment:
I have a bad feeling about this. If a `ContestSubmission` is lost, nothing happens, but if a `SubmissionSource` is lost, all hell breaks loose. Suggest that you put the and the previous block in `with transaction.atomic():`.